### PR TITLE
Test: calling `study.tell` from another process 

### PR DESCRIPTION
**Pull request by Abelarm**
Added a test for checking the support of `study.tell` being called from another process 

## Motivation
Requested in #4027 

## Description of the changes
Added a new test for calling the `study.tell` from another process, in this case the other process was the `cli` option of optuna, tried to replicate part of the test `test_tell` from `test_cli.py`

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@knshnb @contramundum53 Could you review this PR?

**Comment by not522** [comment]
Could you test the `Study.tell` method instead of the command line interface?

**Comment by Abelarm** [comment]
> Could you test the `Study.tell` method instead of the command line interface?

Yes sure I will!

**Comment by Abelarm** [comment]
@not522 Did it! Is this more or less what you were looking for?

**Comment by not522** [comment]
Thank you for your update! It matches my motivation. In addition, could you add a test for `tell` with the trial number?

File: tests/study_tests/test_study.py [hunk 1]
```diff
@@ -1520,3 +1522,39 @@ def objective(trial: Trial) -> float:
...
+
+        assert len(study.trials) == 2
+        assert study.best_trial.state == TrialState.COMPLETE
+        assert study.best_value == 1.2
+
+        try:
+            pool.starmap(_process_tell, [(study, trial0, 1.2)])
+            assert False  # Should fail because the trial0 is already finished
+        except RuntimeError:
+            assert True
```
[the remark this answers is not in the document]
Comment (Abelarm) [reply tests/study_tests/test_study.py:1560]: That's definitely a nicer way to do it, thank you

**Review by knshnb — APPROVED** [review]
Thanks for the update! LGTM.
@not522 Could you take a brief look if you have time? (because the content has been changed since @contramundum53 approved)

**Review by not522 — COMMENTED** [review]
Thank you for the update! It's almost LGTM. I suggest cosmetic changes.

File: tests/study_tests/test_study.py [hunk 2]
```diff
@@ -1520,3 +1522,37 @@ def objective(trial: Trial) -> float:
         study.enqueue_trial(params={"x": 1}, skip_if_exists=False)
         summaries = get_all_study_summaries(study._storage, include_best_trial=True)
         assert summaries[0].datetime_start is not None
+
+
+def _process_tell(study: Study, trial: Union[Trial, int], values: float) -> None:
+    study.tell(trial, values)
+
+
+def test_study_tell_process() -> None:
```
Comment (knshnb) [inline tests/study_tests/test_study.py:1531]:
```suggestion
def test_tell_from_another_process() -> None:
```
A nit.

**Review by contramundum53 — APPROVED** [review]
LGTM.

**Review by not522 — CHANGES_REQUESTED** [review]
LGTM. Could you resolve conflicts?

**Review by not522 — APPROVED** [review]

**Comment by knshnb** [comment]
@not522 Thanks for resolving the conflict.
@Abelarm Thank you for the contribution to Optuna! Let me merge this PR.

**Pull request merged**